src/main/java/org/candlepin/resource/CloudRegistrationResource.java

```diff
@@ -83,4 +156,149 @@ private CloudRegistrationData getCloudRegistrationData(CloudRegistrationDTO clou
+
+        String cloudInstanceId = authResult.getCloudInstanceId();
+        AnonymousCloudConsumer existingAnonConsumer = anonymousCloudConsumerCurator
+            .getByCloudInstanceId(cloudInstanceId);
+
+        String ownerKey = authResult.getOwnerKey();
+
+        // verify that the owner exists upstream and is entitled
+        boolean createAnonConsumer = false;
+        if (ownerKey == null || ownerKey.isBlank() || !authResult.isEntitled()) {
+            CloudAccountOrgSetupJobConfig jobConfig = CloudAccountOrgSetupJob.createJobConfig()
+                .setCloudAccountId(authResult.getCloudAccountId())
+                .setCloudOfferingId(authResult.getOfferId())
+                .setCloudProvider(authResult.getCloudProvider());
+
+            try {
+                this.jobManager.queueJob(jobConfig);
+            }
+            catch (JobException e) {
+                String errMsg = this.i18n.tr("An unexpected exception occurred " +
+                    "while scheduling job \"{0}\"", jobConfig.getJobKey());
+                log.error(errMsg, e);
+                throw new IseException(errMsg, e);
+            }
+
+            if (existingAnonConsumer == null) {
+                createAnonConsumer = true;
+            }
+
+            log.info(
+                "Cloud account org setup job created for account {} {} using offer {}",
+                authResult.getCloudProvider().shortName(), authResult.getCloudAccountId(),
+                authResult.getOfferId());
+        }
+
+        // Impl note: It is possible the owner and subscription exists through the adapters, but
+        // the owner and pool do not yet exist in Candlepin. If that is the case we will want to create
+        // an anonymous cloud consumer record and provide an anonymous registration token.
+        String anonymousConsumerUuid = existingAnonConsumer == null ? null : existingAnonConsumer.getUuid();
+        boolean ownerAndPoolExists = doesOwnerAndPoolExist(ownerKey, authResult.getProductId());
+        if ((!ownerAndPoolExists && existingAnonConsumer == null) || createAnonConsumer) {
+            AnonymousCloudConsumer createdAnonConsumer = new AnonymousCloudConsumer()
+                .setCloudAccountId(authResult.getCloudAccountId())
+                .setCloudInstanceId(authResult.getCloudInstanceId())
+                .setProductId(authResult.getProductId())
+                .setCloudProviderShortName(authResult.getCloudProvider().shortName());
+
+            createdAnonConsumer = anonymousCloudConsumerCurator.create(createdAnonConsumer);
+            anonymousConsumerUuid = createdAnonConsumer.getUuid();
+
+            log.info("Anonymous consumer created for instance {} using cloud account {} {}",
+                authResult.getCloudInstanceId(), authResult.getCloudProvider().shortName(),
+                authResult.getCloudAccountId());
+        }
+
+        String token = anonymousConsumerUuid == null ?
```

Comment:
This bit confuses me somewhat. This flow is for requesting a v2 anon auth token, but it's possible we kick back a v1 token which is only good for doing passwordless registration.

What's the client flow expected to be in this case? Depending on the output they will go into normal registration or request the anon content cert?

Reply:
So this isn't requesting a v2 anon auth token, it is only asking to authenticate the cloud instance metadata. So, we still have to handle the case already existing workflows where we don't need to generate an anonymous token because we don't need to do any additional processing on the backend (create orgs, subscriptions, etc). So, we send the standard token because the client can use it for the passwordless registration. 

> What's the client flow expected to be in this case? Depending on the output they will go into normal registration or request the anon content cert?

Correct! The client will now have two different workflows based on the token type. If the token type is standard then they can use the existing workflow and do a standard registration with the token. If the token type is anonymous then they will have to get the anonymous content certificate until they can fully register after all backend processing is completed.

Reply:
> So this isn't requesting a v2 anon auth token, it is only asking to authenticate the cloud instance metadata. 

I don't follow. This particular end point is used to fetch a token, when is then used to either register directly (v1) or fetch an anon SCA cert while waiting to register (v2) (feel free to correct or add context there); providing the cloud instance metadata is just a means to that end.

What struck me as odd is that we opted to have the client specify which version of the logic they're invoking via the version parameter, but depending on the result of the upstream processing of it, would kick back a different token entirely. That looks to have been corrected in the latest update, so the point is probably moot now.

> The client will now have two different workflows based on the token type.

I would hope that the flow has already been determined by the client before the request is made. These feel different enough that the difference in token should occur at the behest of the client, rather than making the client react to our output. Again, this looks to have been addressed, so my initial confusion or concern may no longer be warranted.

Reply:
ARRRGH, Githuuuuub! The link went to the `cloudRegister` method, not the `processAdapterAuthResult` method, and I based the entirety of my response on what I tricked myself into thinking was a refactor of the logic.

Looking at the *actual* line in question again, I'm still a bit confused as to the intended usage of both tokens here. Given that their scope and usage are completely different, is the client already aware of what token they expect to get back? Are we setting the client up for failure here if the client is expecting an anon token and we give them back the standard token?

I wasn't expecting this functionality to have overlap with the v1/standard tokens at all.

Reply:
When the client makes a request to GET /cloud/authorize the client can't know what type of token to expect because that is dependent on the state of the server. However, the future requests after the call to GET /cloud/authorize is dependent on the token type that is returned. This is what I believe the differences in the workflows:

**Standard Token Workflow (Auto reg v1)**
1. Request to GET /cloud/authorize and receives a standard token with owner key
2. Request to POST /consumers to register system to the owner and get identity certificate
3. Request to GET /consumers/{consumer_uuid}/content_access to get content access certificate

**Anonymous Token Workflow**
1. Request to GET /cloud/authorize and receives an anonymous token with anonymous consumer uuid
2. Request to GET /consumers/{consumer_uuid}/content_access to get anonymous content access certificate